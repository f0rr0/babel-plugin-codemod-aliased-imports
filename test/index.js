import path from "path";
import { transform, transformFileSync } from "@babel/core";
import plugin from "../src";

const withOptions = (options = {}) => ({
  babelrc: false,
  plugins: [[plugin, options]]
});

const getFixture = filename =>
  path.join(__dirname, `./fixtures/${filename}.js`);

describe("codemod-aliased-imports", () => {
  it("throws if alias not provided", () => {
    expect(() => transform("", withOptions())).toThrow(
      "babel-plugin-codemod-aliased-imports expected an `alias` option of type string. Received undefined instead."
    );
  });

  it("throws if path not provided", () => {
    expect(() =>
      transform(
        "",
        withOptions({
          alias: ""
        })
      )
    ).toThrow(
      "babel-plugin-codemod-aliased-imports expected an `path` option of type string. Received undefined instead."
    );
  });

  it("handles imports correctly", () => {
    expect(
      transformFileSync(
        getFixture("sibling"),
        withOptions({
          alias: "~",
          path: "./test/fixtures/foo"
        })
      ).code
    ).toMatchSnapshot();

    expect(
      transformFileSync(
        getFixture("ancestor"),
        withOptions({
          alias: "~",
          path: "./"
        })
      ).code
    ).toMatchSnapshot();

    expect(
      transformFileSync(
        getFixture("child"),
        withOptions({
          alias: "~",
          path: "./test/fixtures/bar"
        })
      ).code
    ).toMatchSnapshot();
  });
});
