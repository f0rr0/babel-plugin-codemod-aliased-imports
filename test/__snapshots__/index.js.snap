// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`codemod-aliased-imports handles imports correctly 1`] = `"import foo from \\"~\\";"`;

exports[`codemod-aliased-imports handles imports correctly 2`] = `"import foo from \\"~/foo/bar\\";"`;

exports[`codemod-aliased-imports handles imports correctly 3`] = `"import baz from \\"~/foo/baz\\";"`;

exports[`codemod-aliased-imports handles imports correctly 4`] = `"import { foo as bar } from \\"~/foo/baz\\";"`;
